src/NUnitEngine/nunit.engine/Drivers/NUnit3FrameworkDriver.cs

```diff
@@ -120,7 +121,20 @@ public string Run(ITestEventListener listener, string filter)
             CallbackHandler handler = new RunTestsCallbackHandler(listener);
 
             log.Info("Running {0} - see separate log file", Path.GetFileName(_testAssemblyPath));
-            CreateObject(RUN_ACTION, _frameworkController, filter, handler);
+
+            try
+            {
+                CreateObject(RUN_ACTION, _frameworkController, filter, handler);
+            }
+            catch (TargetInvocationException ex)
+            {
+                if (ex.InnerException != null)
+                {
+                    throw new NUnitEngineException(ex.InnerException.Message);
+                }
+
+                throw;
+            }
```

Comment:
What problem are you trying to solve here? Why is this better than letting the exception be handled higher in the chain of calls? The driver doesn't have any special knowledge to help it deal with this exception as far as I can see.